impl/src/checks.rs

```diff
@@ -37,19 +38,67 @@ const MAX_DISPLAYED_MISSING_RESOLVE_PACKAGES: usize = 5;
 
 /// Ensure that the given Metadata is valid and ready to use for planning.
 pub fn check_metadata(
-  metadata: &Metadata,
+  raze_metadata: &RazeMetadata,
   settings: &RazeSettings,
   bazel_workspace_root: &Path,
 ) -> Result<()> {
   // Check for errors
-  check_resolve_matches_packages(metadata)?;
+  check_resolve_matches_packages(&raze_metadata.metadata)?;
 
   if settings.genmode == GenMode::Vendored {
-    check_all_vendored(metadata, settings, bazel_workspace_root)?;
+    check_all_vendored(&raze_metadata.metadata, settings, bazel_workspace_root)?;
+  }
+
+  // Check for incomplete lockfiles
+  if let Err(err) =
+    check_lockfile_for_missing_checksums(&raze_metadata.metadata, &raze_metadata.checksums)
+  {
+    eprintln!("WARNING: {}", err);
```

Comment:
This is a non-fatal warning?

Comment in code why?

Reply:
It wasn't originally fatal so I'm maintaining that behavior. We've just moved from silently writing repository definitions with missing `sha256` fields to informing users about it. Do you think it should be an error?

Reply:
Hmm. I don't feel strongly either way. I just thought this looked weird. 

Current state sgtm.